src/main/java/com/scalyr/integrations/kafka/AddEventsClient.java

```diff
@@ -128,7 +128,8 @@ public AddEventsClient(String scalyrUrl, String apiKey, long addEventsTimeoutMs,
     this.initialBackoffDelayMs = initialBackoffDelayMs;
     this.compressor = compressor;
     this.runWithDelay = runWithDelay != null ? runWithDelay : (delayMs, task) -> retryExecutor.schedule(task, delayMs, TimeUnit.MILLISECONDS);
-    this.requestBuilder = HttpWrapper.asyncHttpClient.preparePost(buildAddEventsUri(scalyrUrl));
+    final String addEventsUri = buildAddEventsUri(scalyrUrl);
+    this.requestBuilder = HttpResource.acquire().preparePost(addEventsUri);
```

Comment:
Why declare the `addEventsUri` variable?  Maybe change it back to just inlining the call?

Reply:
`buildAddEventsUri` may throw an `Exception`.  In this case the `AsyncHttpClient` can't be closed b/c the Exception happens in the constructor and the `AddEventsClient` instance is not constructed.

Reply:
Ok, that makes sense but it is not clear from the code that this type of extra protection is necessary.  I could see someone going along and saying "Oh, we should clean this up."  And, in general, this is a problem for anything in this constructor that could throw an exception.  So, let's be defensive and code it up that way from the start.

Something like:
```
   AsyncHttpClient client = HttpResource.acquire();
  try {
     ...
     this.requestBuilder = client.preparePost(buildAddEventsUri(scalyrUrl));
     client = null;   // Must be set only when no exceptions could be thrown.
  } finally {
    if (client != null) HttpResource.release();
  }

```